app/bundles/CoreBundle/Helper/Chart/ChartQuery.php

```diff
@@ -70,9 +74,11 @@ class ChartQuery extends AbstractChart
      */
     public function __construct(Connection $connection, \DateTime $dateFrom, \DateTime $dateTo, $unit = null)
     {
+        $this->unit = $this->getTimeUnitFromDateRange($dateFrom, $dateTo);
+        $this->isTimeUnit = (in_array($unit, ['H', 'i', 's']));
         $this->setDateRange($dateFrom, $dateTo);
         $this->connection = $connection;
-        $this->unit       = !$unit ? $this->getTimeUnitFromDateRange() : $unit;
```

Comment:
This might cause troubles when a user want a different unit than the calculated one. It may happen via a API call for example.